torch/csrc/autograd/profiler.h

```diff
@@ -112,6 +112,31 @@ struct TORCH_API ProfilerConfig {
   ProfilerState state;
   bool report_input_shapes;
   bool profile_memory;
+
+  // Returns IValues corresponding to ProfilerConfig struct, to be used for
+  // serialization.
+  at::IValue toIValues() const {
+    c10::impl::GenericList eventIValueList(at::AnyType::get());
+    eventIValueList.push_back(at::IValue(static_cast<int64_t>(state)));
+    eventIValueList.push_back(at::IValue(report_input_shapes));
+    eventIValueList.push_back(at::IValue(profile_memory));
+    return eventIValueList;
+  }
+
+  // Reconstructs a ProfilerConfig from IValues given by toIValues.
+  static ProfilerConfig fromIValue(at::IValue profilerConfigIValue) {
+    TORCH_INTERNAL_ASSERT(
+      profilerConfigIValue.isList(),
+      "Expected IValue to containt ype c10::impl::GenericList");
+    auto ivalues = profilerConfigIValue.toList();
```

Comment:
also `TORCH_INTERNAL_ASSERT` on the number of elements in the list?

Reply:
Updated